services/files/run_pull.go

```diff
@@ -129,18 +131,11 @@ func (s service) downloadFileTranslations(params PullParams, file sdkfile.File)
 
 	for _, locale := range translations {
 		var complete int64
-
 		if locale.CompletedStringCount > 0 {
-			complete = int64(
-				100 *
-					float64(locale.CompletedStringCount) /
-					float64(status.TotalStringCount),
-			)
-		}
-
-		if percents > 0 {
-			if complete < percents {
-				continue
+			if total := locale.CompletedStringCount + locale.AuthorizedStringCount; total > 0 {
+				complete = int64(
```

Comment:
Pls add unit tests. The formula is tricky, especially rounding. Proposed tests cover this and other cases.